spring-cloud-netflix-zuul/src/main/java/org/springframework/cloud/netflix/zuul/filters/SimpleRouteLocator.java

```diff
@@ -109,26 +118,34 @@ protected Route getSimpleMatchingRoute(final String path) {
 					+ RequestUtils.isZuulServletRequest());
 		}
 
-		String adjustedPath = adjustPath(path);
+		String adjustedPath = adjustPath(request.getPath());
 
-		ZuulRoute route = getZuulRoute(adjustedPath);
+		ZuulRoute route = getZuulRoute(adjustedPath, request.getMethod());
 
 		return getRoute(route, adjustedPath);
 	}
 
-	protected ZuulRoute getZuulRoute(String adjustedPath) {
+	protected ZuulRoute getZuulRoute(String adjustedPath, HttpMethod method) {
 		if (!matchesIgnoredPatterns(adjustedPath)) {
-			for (Entry<String, ZuulRoute> entry : getRoutesMap().entrySet()) {
+			for (Entry<String, Map<HttpMethod, ZuulRoute>> entry : getRoutesMap().entrySet()) {
 				String pattern = entry.getKey();
 				log.debug("Matching pattern:" + pattern);
 				if (this.pathMatcher.match(pattern, adjustedPath)) {
-					return entry.getValue();
+					Map<HttpMethod, ZuulRoute> methods = entry.getValue();
+					return methods.entrySet().stream().filter(e -> matchesMethod(method, e)).findAny().map(Entry::getValue).orElse(null);
 				}
 			}
 		}
 		return null;
 	}
 
+	private boolean matchesMethod(HttpMethod method, Entry<HttpMethod, ZuulRoute> zuulRoute) {
+		if (method == null) {
```

Comment:
is this even possible?

Reply:
It is, as now the RouteLocator interface defines 2 getMatchingRoute methods. The first one is the original one which I made deprecated and the new one is a *default* method with the following implementation:
`return getMatchingRoute(request.getPath());`
meaning that we won't have method here.

This means that if someone already had a special RouteLocator implementation, we don't want to break that but just falling back to it's original implementation.

The reason is that I just didn't want to introduce any breaking API changes, but if you are OK with that, I can remove the original method from the interface.

Reply:
This is going into the 2.0.x release so this is the time to make breaking API changes

Reply:
Okay, then I'll introduce the breaking change.

However, does it make sense to go with this implementation for 1.x (next release)?

When do you guys plan to release 2.0?